File: backend/src/main/java/com/backend/petplace/global/jwt/CustomUserDetailsService.java
package com.backend.petplace.global.jwt;

import com.backend.petplace.domain.user.entity.User;
import com.backend.petplace.domain.user.repository.UserRepository;
import com.backend.petplace.global.exception.BusinessException;
import com.backend.petplace.global.response.ErrorCode;
import lombok.RequiredArgsConstructor;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class CustomUserDetailsService implements UserDetailsService {

  private final UserRepository userRepository;

  @Override
  public UserDetails loadUserByUsername(String idAsString) throws UsernameNotFoundException {
    Long userId = Long.parseLong(idAsString);
    User user = userRepository.findById(userId)
        .orElseThrow(() -> new BusinessException(ErrorCode.NOT_FOUND_MEMBER));
    return new CustomUserDetails(user);
  }
}

File: backend/src/main/java/com/backend/petplace/global/jwt/CustomUserDetailsService.kt
package com.backend.petplace.global.jwt

import com.backend.petplace.domain.user.repository.UserRepository
import com.backend.petplace.global.exception.BusinessException
import com.backend.petplace.global.response.ErrorCode
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.stereotype.Service

@Service
class CustomUserDetailsService(
    private val userRepository: UserRepository
) : UserDetailsService {

    override fun loadUserByUsername(idAsString: String): UserDetails {
        val userId = idAsString.toLongOrNull()
            ?: throw BusinessException(ErrorCode.NOT_FOUND_MEMBER)

        val user = userRepository.findById(userId)
            .orElseThrow { BusinessException(ErrorCode.NOT_FOUND_MEMBER) }

        return CustomUserDetails(user)
    }
}